### PR TITLE
Single package for backend interfaces

### PR DESCRIPTION
**Pull request by osipov-mit**
- Saving blockHash for every event
- Field timestamp for program and message entity
- Package with interfaces that are used by `api-gateway`, `data-storage`, `events-listener`
- Fix Dockerfiles

### CODE REVIEW

File: website/data-storage/src/consumer/consumer.controller.ts [hunk 1]
```diff
@@ -9,57 +21,56 @@ export class ConsumerController {
   constructor(private readonly consumerService: ConsumerService) {}
 
   @MessagePattern('events')
-  async addEvent(@Payload() payload) {
-    logger.log(payload.value, 'AddEvent');
-    const genesis = payload.headers.genesis;
+  async addEvent(@Payload() payload: AddEventKafkaPayload<Keys, any>) {
+    // const genesis = payload.headers.genesis;
```
Comment (xhjkl) [inline website/data-storage/src/consumer/consumer.controller.ts:25]: Stale code?

**Review by xhjkl — APPROVED** [review]

File: website/data-storage/src/consumer/consumer.controller.ts (hunk 1 above)
Comment (osipov-mit) [reply website/data-storage/src/consumer/consumer.controller.ts:25]: yeah, thank you

File: website/interfaces/src/kafka-payload.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface KafkaPayload<T> {
+  headers: any;
```
Comment (codev0) [inline website/interfaces/src/kafka-payload.ts:2]: if it is http headers, we could describe its type as `Record<string, string>`

**Review by nikitayutanov — APPROVED** [review]

File: website/interfaces/src/kafka-payload.ts (hunk 2 above)
Comment (osipov-mit) [reply website/interfaces/src/kafka-payload.ts:2]: No, it isn't

**Pull request merged**